src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/AdminCourtAreaOfLawEndpointTest.java

```diff
@@ -0,0 +1,158 @@
+
+        final var response = doPutRequest(
+            AYLESBURY_COURT_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + authenticatedToken),
+            updatedJson
+        );
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<AreaOfLaw> updatedCourtAreaOfLaw = response.body().jsonPath().getList(
+            ".",
+            AreaOfLaw.class
+        );
+        assertThat(updatedCourtAreaOfLaw).containsExactlyElementsOf(expectedCourtAreasOfLaw);
+
+        //clean up by removing added record
+        final var cleanUpResponse = doPutRequest(
+            AYLESBURY_COURT_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + authenticatedToken),
+            originalJson
+        );
+        assertThat(cleanUpResponse.statusCode()).isEqualTo(OK.value());
+
+        final List<AreaOfLaw> cleanCourtAreaOfLaw = cleanUpResponse.body().jsonPath().getList(
+            ".",
+            AreaOfLaw.class
+        );
+        assertThat(cleanCourtAreaOfLaw).containsExactlyElementsOf(currentCourtAreasOfLaw);
+    }
+
+    @Test
+    public void shouldRequireATokenWhenUpdatingAreaOfLawForTheCourt() throws JsonProcessingException {
+        final var response = doPutRequest(AYLESBURY_COURT_AREAS_OF_LAW_PATH, getTestAreaOfLawJson());
+        assertThat(response.statusCode()).isEqualTo(UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void shouldBeForbiddenForUpdatingAreaOfLawForTheCourt() throws JsonProcessingException {
+        final var response = doPutRequest(
+            AYLESBURY_COURT_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + forbiddenToken), getTestAreaOfLawJson()
+        );
+        assertThat(response.statusCode()).isEqualTo(FORBIDDEN.value());
+    }
+
+    @Test
+    public void shouldBeNotFoundForUpdateWhereAreaOfLawDoesNotExist() throws JsonProcessingException {
+        final Response response = doPutRequest(AYLESBURY_COURT_AREAS_OF_LAW_NOT_FOUND_PATH,
+                                               Map.of(AUTHORIZATION, BEARER + authenticatedToken),
+                                               getTestAreaOfLawJson()
+        );
+        assertThat(response.statusCode()).isEqualTo(NOT_FOUND.value());
+    }
+
+    /************************************************************* Shared utility methods. ***************************************************************/
+    private List<AreaOfLaw> getCurrentCourtAreasOfLaw() {
+        final var response = doGetRequest(
+            AYLESBURY_COURT_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + authenticatedToken)
+        );
+        return response.body().jsonPath().getList(".", AreaOfLaw.class);
+    }
+
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    private List<AreaOfLaw> updateCourtAreasOfLaw(final List<AreaOfLaw> courtsAreaOfLaw) {
+        List<AreaOfLaw> updatedAreaOfLawAuthorities = new ArrayList<>(courtsAreaOfLaw);
```

Comment:
@ashwini-mv What does it mean by "updatedAreaOfLawAuthorities"? The variable name should be updated